R/filter-dm.R

```diff
@@ -9,10 +9,19 @@
 #'
 #' @details `cdm_filter()` allows you to set one or more filter conditions for one table
 #' of a [`dm`] object. These conditions will be stored in the [`dm`] for when they are needed.
-#' Once executed, the filtering the will affect all tables connected to the filtered one by
-#' foreign key constraints, leaving only the rows with the corresponding key values. The filtering
-#' implicitly takes place, once a table is requested from the [`dm`] by using one of `tbl()`, `[[.dm()`, `$.dm()`.
+#' The conditions are only evaluated in one of the following scenarios:
+#' 1. Calling `cdm_apply_filters()` or `compute()` (method for `dm` objects) on a `dm`: each filtered table potentially
+#' reduces the rows of all other tables connected to it by foreign key relations (cascading effect), only leaving the rows
+#' with the corresponding key values. Isolated (in terms of foreign key constraints) and unfiltered tables are left untouched.
+#' This results in a new `dm` class object.
+#' 1. Calling one of `tbl()`, `[[.dm()`, `$.dm()`: the remaining rows of the requested table are calculated based on the
+#' filter conditions and the foreign key conditions (similar to 1. but only for one table)
 #'
+#' It should be mentioned, that if an UNFILTERED table had a potential effect on other tables -- since it lacks key
```

Comment:
This is confusing and a bit too detailed, maybe add an example that highlights this effect?

Reply:
I tried explaining it a bit better now and added an example (1f9c06975719cadde3b2a1badd44be03d9b38ff8).

We could also leave the explanatory text away and let the example speak for itself(?)

Reply:
I'm in favor of removing the text and only explain by example.